### PR TITLE
fix some problems during balance

### PR DESCRIPTION
**Pull request by critical27**
Fix some problems during test:
1. Previously we update when BalanceProcessor has return, but the part alloc probably would be updated later, so some machine won't `loadData` in MetaClient. So I update meta when a BalanceTask has done `updateMeta`.
2. When we `balance remove xxx`, the src of BalanceTask could be alive, but previously we regard them as dead, which would cause some part was not removed.
3. We should forbid `balance leader` during `balance data`.
4. We don't need to `refreshCache` when add a part.

### CODE REVIEW

File: src/meta/processors/admin/BalanceTask.cpp [hunk 1]
```diff
@@ -152,6 +152,14 @@ void BalanceTask::invoke() {
                 if (!resp.ok()) {
                     LOG(INFO) << taskIdStr_ << "Update meta failed, status " << resp;
                     ret_ = Result::FAILED;
+                } else if (kv_ != nullptr) {
+                    if (LastUpdateTimeMan::update(kv_, time::WallClock::fastNowInMilliSec()) !=
```
Comment (dangleptr) [inline src/meta/processors/admin/BalanceTask.cpp]: Each task finished will result in the meta info updated?

Comment (critical27) [reply src/meta/processors/admin/BalanceTask.cpp]: Do you have better suggestion? How about when BalancePlan::invoke is done?

Comment (dangleptr) [reply src/meta/processors/admin/BalanceTask.cpp]: Sounds good.

File: src/meta/processors/partsMan/ListHostsProcessor.cpp [hunk 2]
```diff
@@ -74,7 +74,7 @@ Status ListHostsProcessor::allHostsWithStatus() {
     }
 
     // get hosts which have send heartbeat recently
-    auto activeHosts = ActiveHostsMan::getActiveHosts(kvstore_, FLAGS_heartbeat_interval_secs + 1);
+    auto activeHosts = ActiveHostsMan::getActiveHosts(kvstore_, FLAGS_heartbeat_interval_secs * 2);
```
Comment (darionyaphet) [inline src/meta/processors/partsMan/ListHostsProcessor.cpp]: Actually I'm not sure why heartbeat_interval_secs * 2 after talk with dooble offline, maybe more safe

Comment (dangleptr) [reply src/meta/processors/partsMan/ListHostsProcessor.cpp]: Yes,  the purpose is to make the active hosts  returned more stable.

**Pull request merged**